src/background/extension.ts

```diff
@@ -54,30 +55,49 @@ export class Extension {
         this.awaiting = [];
     }
 
-    isEnabled() {
+    private alarmListener = (alarm): void => {
+        if (alarm.name === Extension.ALARM_NAME) {
+            this.handleAutoCheck();
+        }
+    };
+
+    isEnabled(): boolean {
+        if (this.isEnabledNow !== null) {
+            return this.isEnabledNow;
+        }
+
         const {automation} = this.user.settings;
-        if (automation === 'time') {
-            const now = new Date();
-            return isInTimeInterval(now, this.user.settings.time.activation, this.user.settings.time.deactivation);
-        } else if (automation === 'system') {
-            if (isFirefox) {
-                // BUG: Firefox background page always matches initial color scheme.
-                return this.wasLastColorSchemeDark == null
-                    ? isSystemDarkModeEnabled()
-                    : this.wasLastColorSchemeDark;
-            }
-            return isSystemDarkModeEnabled();
-        } else if (automation === 'location') {
-            const latitude = this.user.settings.location.latitude;
-            const longitude = this.user.settings.location.longitude;
-
-            if (latitude != null && longitude != null) {
-                const now = new Date();
-                return isNightAtLocation(now, latitude, longitude);
+
+        let goodTime: TimeCheck = null;
```

Comment:
```suggestion
        let timingInformation: TimeCheck = null;
```

Or any other variable 😄

Reply:
Done!